### PR TITLE
Slim 2869 secret mgmt tests

### PR DESCRIPTION
**Pull request by jandik**
Code changes due to test issues reported (secrets with expired keys were not returned)

### CODE REVIEW

File: osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/services/SecretManagementService.java [hunk 1]
```diff
@@ -53,6 +54,7 @@
  * by reencrypting the AES-encrypted secrets to RSA.
  */
 @Service
+@Slf4j
```
Comment (carlogeenen) [inline osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/services/SecretManagementService.java:57]: I think this entire file could be out of this PR. I assume this the result of undoing the added log statements?

**Review by carlogeenen — COMMENTED** [review]
I think SecretManagementService changes could be left out.

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java [hunk 2]
```diff
@@ -60,10 +60,10 @@ public HsmEncryptionProvider(final File keyStoreFile) {
     public byte[] decrypt(final EncryptedSecret secret, final String keyReference) {
         byte[] decryptedSecret = super.decrypt(secret, keyReference);
         if (decryptedSecret.length > KEY_LENGTH) {
-            final byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecret, 0,
-                    decryptedSecret.length-16);
-            LOGGER.trace("Truncating decrypted key from " + Hex.encodeHexString(decryptedSecret) + " to " +
-                            Hex.encodeHexString(truncatedDecryptedSecretBytes));
+            final byte[] truncatedDecryptedSecretBytes = Arrays
+                    .copyOfRange(decryptedSecret, 0, decryptedSecret.length - 16);
```
Comment (carlogeenen) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java:64]: replace 16 by KEY_LENGTH

Comment (jandik) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java:64]: Yeah, I talked to Erik about this some time ago; the way it is programmed here is confusing and only works with particular key sizes and block sizes, which are fortunately always ok. Rewritten this to something more logical. Retest on HSM is required!

**Review by carlogeenen — APPROVED** [review]

**Review by carlogeenen — APPROVED** [review]

**Pull request merged**